src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -121,3 +122,19 @@ float64_t GaussianKernel::distance(int32_t idx_a, int32_t idx_b) const
 {
 	return ShiftInvariantKernel::distance(idx_a, idx_b)/get_width();
 }
+
+SGMatrix<float64_t> 
+GaussianKernel::sample_spectral_density(PRNG& prng, int32_t dim_output, int32_t dim_input_space) const
+{
+	NormalDistribution<float64_t> normal_dist;
+	SGMatrix<float64_t> sampled_kernel(dim_output, dim_input_space);
+	const auto width = get_width();
+	const auto std_dev = std::sqrt(2.0 / width);
+	std::generate(
+	    sampled_kernel.begin(), sampled_kernel.end(),
+	    [this, &normal_dist, &std_dev, &prng]() {
+		    return std_dev * normal_dist(prng);
+	    });
+
```

Comment:
https://stackoverflow.com/questions/31410209/perfectly-capturing-a-perfect-forwarder-universal-reference-in-a-lambda/31410880#31410880

Reply:
I think I've lost my sense of reasoning reading that code. Are we sure we want to use that bazooka here?

Reply:
At the risk of being wrong, how about this?
```c++
 sample_spectral_density(PRNG&& prng .....)
{
   /*Code*/
   [this, &normal_dist, &std_dev, &prng]() {
		    return std_dev * normal_dist(prng);
	    });
}
```
This works fine, and why are we adamant on rval ref, shouldn't lval ref do the trick too?

Reply:
I guess what @karlnapf means is `perfect forward` the `prng` to `normal_dist`. but in the current implementation of `NormalDistribution`, `operator()` is not a `perfect forward` function.
https://github.com/shogun-toolbox/shogun/blob/a075792c536d67e295927577d12b7bf335565550/src/shogun/mathematics/NormalDistribution.h#L37-L43

Reply:
That is a good point!